packages/dashboard/src/customization/widgets/lineSymbol/component.test.tsx

```diff
@@ -0,0 +1,221 @@
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import LineWidgetComponent from './component';
+import { Provider } from 'react-redux';
+import { DashboardStore, configureDashboardStore } from '~/store';
+import { DashboardState } from '~/store/state';
+import { LineProperties, LineWidget } from '../types';
+import { act } from 'react-dom/test-utils';
+
+const mockWidgetPartial = {
+  id: 'mock-line-widget',
+  type: 'line',
+  x: 0,
+  y: 0,
+  z: 1,
+  width: 7,
+  height: 4,
+};
+
+const TestComponent = (object: { widget: LineWidget; isSelected: boolean }) => {
+  const { widget, isSelected } = object;
+
+  const initialState: Partial<DashboardState> = {
+    dashboardConfiguration: {
+      widgets: [widget],
+      viewport: { duration: '5m' },
+    },
+    selectedWidgets: isSelected ? [widget] : [],
+  };
+
+  return (
+    // sets up all values for the test
+    <Provider store={configureDashboardStore(initialState)}>
+      <DndProvider backend={HTML5Backend}>
+        <LineWidgetComponent {...widget} />
+      </DndProvider>
+    </Provider>
+  );
+};
+
+describe('LineWidgetComponent', () => {
+  describe('Rendering', () => {
+    [
+      {
+        properties: {
+          lineStyle: 'solid',
+          color: 'black',
+          thickness: 5,
+          start: { x: 20, y: 20 },
+          end: { x: 180, y: 180 },
+        } as LineProperties,
+        isSelected: true,
+      },
+      {
+        properties: {
+          lineStyle: 'dashed',
+          color: 'black',
+          thickness: 5,
+          start: { x: 30, y: 20 },
+          end: { x: 180, y: 190 },
+        } as LineProperties,
+        isSelected: true,
+      },
+      {
+        properties: {
+          lineStyle: 'dotted',
+          color: 'black',
+          thickness: 5,
+          start: { x: 25, y: 5 },
+          end: { x: 100, y: 120 },
+        } as LineProperties,
+        isSelected: true,
+      },
+      {
+        properties: {
+          lineStyle: 'solid',
+          color: 'red',
+          thickness: 5,
+          start: { x: 20, y: 20 },
+          end: { x: 180, y: 180 },
+        } as LineProperties,
+        isSelected: true,
+      },
+      {
+        properties: {
+          lineStyle: 'dashed',
+          color: 'blue',
+          thickness: 10,
+          start: { x: 180, y: 175 },
+          end: { x: 20, y: 30 },
+        } as LineProperties,
+        isSelected: true,
+      },
+      {
+        properties: {
+          lineStyle: 'dotted',
+          color: 'green',
+          thickness: 15,
+          start: { x: 180, y: 175 },
+          end: { x: 20, y: 30 },
+        } as LineProperties,
+        isSelected: true,
+      },
+    ].forEach((configuration) => {
+      const widget = { ...mockWidgetPartial, properties: { ...configuration.properties } };
+      const isSelected = configuration.isSelected;
+      it(`should render with ${JSON.stringify(widget)} correctly. The widget ${
```

Comment:
Is this line readable in the console? Can we make the title of the test more succinct?

Reply:
good point. i replaced `widget` with `configuration` since that is more relevant. also shortened the title.